api/requests/scene_items/xx_generated.getsceneitemproperties.go

```diff
@@ -2,22 +2,30 @@
 
 package sceneitems
 
-import requests "github.com/andreykaipov/goobs/api/requests"
+import (
+	requests "github.com/andreykaipov/goobs/api/requests"
+	typedefs "github.com/andreykaipov/goobs/api/typedefs"
+)
 
 /*
 GetSceneItemPropertiesParams represents the params body for the "GetSceneItemProperties" request.
 Gets the scene specific properties of the specified source item.
 Coordinates are relative to the item's parent (the scene or group it belongs to).
 
-Generated from https://github.com/Palakis/obs-websocket/blob/4.7.0/docs/generated/protocol.md#GetSceneItemProperties.
+Generated from https://github.com/Palakis/obs-websocket/blob/4.8.0/docs/generated/protocol.md#GetSceneItemProperties.
 */
 type GetSceneItemPropertiesParams struct {
 	requests.ParamsBasic
 
-	// The name of the source.
-	Item string `json:"item"`
+	Item struct {
```

Comment:
This (and similar) changes actually breaks the library. No matter what `Name` I set, it doesn't seem to be able to retrieve scene items anymore. I assume this is because even if I don't set `Id`, it will still be a valid integer (zero) and used instead of the provided name.

Reply:
I guess one way to solve this is making the `optional` request parameters pointers of their type (e.g. `*int`), and adding the `omitempty` json tag to them.

Reply:
Oh I see :(

I was a bit confused at this diff because I remember explicitly choosing `String` for the `String | Object` type for `item` (see https://github.com/andreykaipov/goobs/pull/5/commits/88018a7a5ffb719bb0378f99138fa0456f500492#diff-36f7fb652cc75e333e8bff2b91e2bb5562e40ff081e5354aacd04d71b6fedaf4R288-R299), but I see now this field still got parsed a struct because that's what `item.name` and `item.id` imply.

In any case, I think adding the `omitempty` option (`json:"blah,omitempty"`) will fix this. Probably should do that for every field in fact, not just the optional ones. I remember telling myself to add that in the past but didn't write it down anywhere. 😅 Thank you for catching this!

Reply:
https://github.com/andreykaipov/goobs/issues/12

Reply:
Oh sorry I commented before refreshing the thread so didn't see your second comment. Do they have to be pointers?

Reply:
They would have to be pointers for `omitempty` to work indeed. Otherwise the JSON encoder can't tell the difference between an unused integer and one that's intentionally set to `0` (or an empty string vs an unset string, etc.)

Reply:
Oh hey, sounds like I remember this wrong:

```
The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.
```

Reply:
Right right. Doing some reading it seems using pointer is sometimes preferred when the zero value of your primitive is a valid value, but I think in this case 0 won't ever be a valid ID, nor a name ever an empty string.